### PR TITLE
MAINT: Migrate to poetry

### PR DESCRIPTION
**Pull request by Kai-Striega**
This is an experimental branch to migrate numpy-financial to poetry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+repository = "https://github.com/numpy/numpy-financial"
+documentation = "https://numpy.org/numpy-financial/latest/#functions"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Financial and Insurance Industry",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
```
Comment (rgommers) [inline pyproject.toml:19]: Minor: double entry for 3.9 here

Comment (rgommers) [reply pyproject.toml:19]: Python 3.12 betas are already out, so you may as well add that line already; nothing will break anymore for pure Python packages like `numpy-financial`.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+[tool.poetry.group.docs.dependencies]
+sphinx = "^7.0"
+numpydoc = "^1.5"
+
+
+[tool.poetry.group.lint.dependencies]
+flake8 = "*"
+
+
+[build-system]
```
Comment (rgommers) [inline pyproject.toml:53]: You may want to move this to the top; this is a standard section, which should come before tool-specific sections.

File: pyproject.toml (hunk 1 above)
Comment (Kai-Striega) [reply pyproject.toml:19]: Thanks for the tip. I'll add it

File: pyproject.toml (hunk 2 above)
Comment (Kai-Striega) [reply pyproject.toml:53]: Thanks, I'll do that. For some reason `poetry init` decided to generate all the poetry specific sections at the top

**Pull request merged**